rd-kt/rd-core/src/jvmMain/kotlin/com/jetbrains/rd/util/PlatformDependent.kt

```diff
@@ -19,7 +19,15 @@ actual fun currentThreadName() : String = Thread.currentThread().run { "$id:$nam
 actual class AtomicReference<T> actual constructor(initial: T) {
     private val impl = AtomicReference(initial)
     actual fun get(): T = impl.get()
-    actual fun getAndUpdate(f: (T) -> T): T = impl.getAndUpdate(f)
+    actual fun getAndUpdate(f: (T) -> T): T {
+        var prev: T
+        var next: T
+        do {
+            prev = impl.get()
+            next = f(prev)
+        } while (!impl.compareAndSet(prev, next))
+        return prev
+    }
```

Comment:
Just as an explanation for interested reader (I know you're reading this!), we currently have a suspicion (unconfirmed yet), that in JDK11 this code could sometimes freeze for several minutes due to changes in `AtomicReference.getAndUpdate`, which now uses `weakCompareAndSet`. So, we're trying to roll it back to JDK8 behavior.

Reply:
🤦‍♂️